crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -625,16 +619,23 @@ impl CheckpointStore {
         let our_proposal = locals.current_proposal.as_ref().unwrap();
 
         if let Ok(Some(contents)) = self.reconstruct_contents(committee, our_proposal) {
+            // Create a new order for the checkpointed contents
+            let old_order: Vec<_> = contents.iter().cloned().collect();
+            let new_order = orderer
+                .get_complete_causal_order(&old_order, self)
```

Comment:
Ok, I have re-introduced this, to guard against it:
```
        // Ensure all transactions included are executed (static property). This should be true since we should not
        // be signing a checkpoint unless we have processed all transactions within it.
        if effects.iter().any(|e| e.is_none()) {
            return Err(SuiError::from(
                "Cannot causally order checkpoint with unexecuted transactions.",
            ));
        }
```